### PR TITLE
Reorg

### PR DESCRIPTION
**Pull request by SatjaSivcev**

### CODE REVIEW

**Comment by knatten** [comment]
I can't really comment on the structural changes here, as I don't think it should be up to me which samples are where etc. So I'll leave that to someone in your team. But will have a look at the technical details.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .vs
 bin
 obj
+build
+/build
```
Comment (knatten) [inline .gitignore]: You shouldn't need both of these, the one on line 5 is already covered by the one on line 4.

**Comment by SatjaSivcev** [comment]
> 
> 
> I can't really comment on the structural changes here, as I don't think it should be up to me which samples are where etc. So I'll leave that to someone in your team. But will have a look at the technical details.

This is where I want oppinion from @torbsorb

**Comment by SatjaSivcev** [comment]
> 
> 
> In general very nice. I like that you did one thing per commit, it makes it very easy to review!
> 
> In the commit "Remove on directory layer", you change the platform from AnyCPU to x64. I think belongs in a separate commit with a message explaining why it was done. You also add some new references to some of the csprojs in that commit, this should also be split out, and needs an explanation.
> 
> Why is there one main solution file, and then some of the samples (but not all) have their own solution files?
> 
> Some of the samples now use Zivid3d.zdf from this repo, and some use MiscObjects.zdf. Is that intentional?
> 
> Remember to squash the revert you did.
> 
> I like the new directory structure!

Why is there one main solution file, and then some of the samples (but not all) have their own solution files? - I wasn't sure what is the preferred way so I hoped to start this discussion. Is it better to have only one main Solution file? Or have one main and also one for each program? Or just one for single solution?

Remember to squash the revert you did. - Yes, this depends on what we decide for above.

Some of the samples now use Zivid3d.zdf from this repo, and some use MiscObjects.zdf. Is that intentional? - Yes. One is ZDF point cloud. The other is File Camera.

**Comment by knatten** [comment]
> Why is there one main solution file, and then some of the samples (but not all) have their own solution files? - I wasn't sure what is the preferred way so I hoped to start this discussion. Is it better to have only one main Solution file? Or have one main and also one for each program? Or just one for single solution?

I think it's simplest to just have one main solution file. Then the user just opens that and can run all the samples. I can't think of a reason why someone would want to open just one sample, and it would be a problem that the others are also available. What do you think?
 
> Some of the samples now use Zivid3d.zdf from this repo, and some use MiscObjects.zdf. Is that intentional? - Yes. One is ZDF point cloud. The other is File Camera.

Ah, that makes sense then! :) Hopefully we will change the file extension for file cameras soon, which should help with these confusions.

**Comment by SatjaSivcev** [comment]
> 
> 
> > Why is there one main solution file, and then some of the samples (but not all) have their own solution files? - I wasn't sure what is the preferred way so I hoped to start this discussion. Is it better to have only one main Solution file? Or have one main and also one for each program? Or just one for single solution?
> 
> I think it's simplest to just have one main solution file. Then the user just opens that and can run all the samples. I can't think of a reason why someone would want to open just one sample, and it would be a problem that the others are also available. What do you think?

I think the same.

> 
> > Some of the samples now use Zivid3d.zdf from this repo, and some use MiscObjects.zdf. Is that intentional? - Yes. One is ZDF point cloud. The other is File Camera.
> 
> Ah, that makes sense then! :) Hopefully we will change the file extension for file cameras soon, which should help with these confusions.

File: .gitignore (hunk 1 above)
Comment (SatjaSivcev) [reply .gitignore]: done

**Comment by SatjaSivcev** [comment]
After a bit of forth and back (as you see above) it is ready for the next round.

I need some input on changing the platform from AnyCPU to x64. I saw that this is how it was done for the basic samples and this is why I changed it. I am not certain whether it is better or not.

**Comment by SatjaSivcev** [comment]
@knatten can you help me understand why we want to change the platform from AnyCPU to x64?

**Comment by knatten** [comment]
Sorry, this one got lost in my backlog.

The Zivid SDK is a 64 bit application. So anything on .net which calls into these native dlls need to be running in a 64 bit process. Changing from `AnyCPU` to `x64` signals to users that these projects are intended to only be run on a 64 bit system.

I think you can just quote that in the commit message.

**Comment by SatjaSivcev** [comment]
Done. Ready for the final review.

**Comment by SatjaSivcev** [comment]
I've removed the build directory. Yes, I've tested all the samples.
The structure is using the C++ samples structure as reference. The changes we want to make we made on the C++ one first.

**Pull request merged**